### PR TITLE
[wip] Move event-store related factories from proophessor to this repo

### PR DESCRIPTION
**Pull request by prolic**
see: https://github.com/prooph/event-store/issues/57

### CODE REVIEW

**Comment by prolic** [comment]
Will be run phpspec + phpunit on this repository or will we have just one test-suite?

File: src/Prooph/EventStore/Service/Stream/EventStoreFactory.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        if (! isset($adapterConfig['type'])) {
+            throw ConfigurationException::configurationError('Missing adapter type configuration in proophessor event_store configuration');
+        }
+
+        $adapterType    = $adapterConfig['type'];
+        $adapterOptions = isset($adapterConfig['options'])? $adapterConfig['options'] : [];
+
+        //Check if we have to use the application wide database connection
+        if ($adapterType == 'Prooph\EventStore\Adapter\Doctrine\DoctrineEventStoreAdapter'
```
Comment (prolic) [inline src/Prooph/EventStore/Service/Stream/EventStoreFactory.php]: We should rely on an adapter factory here, so users are able to provide their own adapters.

**Comment by codeliner** [comment]
Can you submit the PR against 5.0-beta branch? In the future we should use a develop branch. My bad that did not create one already. All changes related to the 5.0 release are included in the 5.0-beta at the moment.

File: src/Prooph/EventStore/Service/Stream/EventStoreFactory.php (hunk 1 above)
Comment (codeliner) [inline src/Prooph/EventStore/Service/Stream/EventStoreFactory.php]: :+1: absolutely. I thought about getting the adapter from the container and fallback to the InMemoryAdapter if no adapter is available. What do you think?

Comment (prolic) [inline src/Prooph/EventStore/Service/Stream/EventStoreFactory.php]: Makes perfect sense. I'll update my PR's.
Am 20.08.2015 18:04 schrieb "Alexander Miertsch" <notifications@github.com>:

**Comment by codeliner** [comment]
> Will be run phpspec + phpunit on this repository or will we have just one test-suite?
ah phpspec, that was an experiment but we now use phpunit > 4.7 and have Prophecy available within phpunit which works really good. So I would like to remove phpspec and continue to work with phpunit only. Unfortunately, we have to migrate the tests from proophessor but I found it useful to write the tests again for prooph/service-bus. I worked a lot with data providers to also test exceptions etc.

File: src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{
+    /**
+     * @param ContainerInterface $container
+     * @return AggregateStreamStrategy
+     */
+    public function __invoke(ContainerInterface $container)
+    {
+        $aggregateTypeStreamMap = [];
+
+        if ($container->has('configuration')) {
```
Comment (codeliner) [inline src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php]: In prooph/service-bus I used `config`. In ZF2 both service names are available. Do you know what is the actual service name in ZF2 and what is only the alias?
Maybe we should check what other frameworks use. `config` is shorter, but maybe to short.

File: src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            $config = $container->get('configuration');
+
+            if (isset($config['proophessor']['event_store']['aggregate_type_stream_map'])) {
```
Comment (codeliner) [inline src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php]: proophessor needs to be changed to prooph

File: src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php (hunk 2 above)
Comment (prolic) [inline src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php]: ZF2's real config name is "Config".

File: src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php (hunk 3 above)
Comment (prolic) [inline src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php]: ok

File: src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                $aggregateTypeStreamMap = $config['proophessor']['event_store']['aggregate_type_stream_map'];
+            }
+        }
+
+        return new AggregateStreamStrategy($container->get('proophessor.event_store'), $aggregateTypeStreamMap);
```
Comment (codeliner) [inline src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php]: Oh, I forgot. The stream strategies have the event store as dependency.
Should we try to handle everything in a single EventStoreFactory, except the adapter set up?

Comment (prolic) [inline src/Prooph/EventStore/Service/Stream/AggregateStreamStrategyFactory.php]: No, let everything be subject to inject by the container.

**Comment by prolic** [comment]
I will create a new PR for the 5.0-beta branch.

**Comment by prolic** [comment]
Can you join the gitter-chat?

**Pull request closed without merging**